### PR TITLE
Add ganon/buildcustom

### LINKED ISSUE #3327: new module: GANON/BUILDCUSTOM
### Is there an existing module for this?

- [X] I have searched for the existing module

### Is there an open PR for this?

- [X] I have searched for existing PRs

### Is there an open issue for this?

- [X] I have searched for existing issues

### Are you going to work on this?

- [X] If I'm planning to work on this module, I added myself to the `Assignees` to facilitate tracking who is working on the module

### PR DESCRIPTION
**Pull request by jfy133**
## PR checklist

Closes #3327 

- [x] This comment contains a description of changes (with reason).
- [x] If you've fixed a bug or added code that should be tested, add tests!
- [x] If you've added a new tool - have you followed the module conventions in the [contribution docs](https://github.com/nf-core/modules/tree/master/.github/CONTRIBUTING.md)
- [x] If necessary, include test data in your PR.
- [x] Remove all TODO statements.
- [x] Emit the `versions.yml` file.
- [x] Follow the naming conventions.
- [x] Follow the parameters requirements.
- [x] Follow the input/output options guidelines.
- [x] Add a resource `label`
- [x] Use BioConda and BioContainers if possible to fulfil software requirements.
- Ensure that the test works with either Docker / Singularity. Conda CI tests can be quite flaky:
  - [x] `PROFILE=docker pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`
  - [x] `PROFILE=singularity pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`
  - [ ] `PROFILE=conda pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`

Note I tried to add support for the (mutually exclusive to `--input`) `--input-file` parameter, but I couldn't find a good way to do this due to a lack of expected standardisation of file inputs for either parameter (having two separate channels sees a `meta` clash, and having two objects in an input tuple became unwieldly and difficult to document in the meta.yml) . I've thus opted for the more 'nextflow' friendly FASTA input only.

I can take a stab at this again in the future if it is requested.

### CODE REVIEW

File: modules/nf-core/ganon/buildcustom/main.nf [hunk 1]
```diff
@@ -0,0 +1,44 @@
+process GANON_BUILDCUSTOM {
+    tag "$meta.id"
+    label 'process_low'
```
Comment (sofstam) [inline modules/nf-core/ganon/buildcustom/main.nf:3]: Might we need `process_medium` or `process_high` here?

File: tests/modules/nf-core/ganon/buildcustom/main.nf [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+nextflow.enable.dsl = 2
+
+include { GANON_BUILDCUSTOM } from '../../../../../modules/nf-core/ganon/buildcustom/main.nf'
+
+workflow test_ganon_buildcustom {
+
+    input = [
+        [ id:'test', single_end:false ], // meta map
+        file(params.test_data['sarscov2']['genome']['genome_fasta'], checkIfExists: true),
+        []
```
Comment (jfy133) [inline tests/modules/nf-core/ganon/buildcustom/main.nf:12]:
```suggestion
```

File: modules/nf-core/ganon/buildcustom/main.nf [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+        $taxonomy_args \\
+        $genome_size_args \\
+        $args
+
+    cat <<-END_VERSIONS > versions.yml
+    "${task.process}":
+        ganon: \$(echo \$(ganon --version 2>1) | sed 's/.*ganon //g')
+    END_VERSIONS
+    """
+}
```
Comment (nvnieuwk) [inline modules/nf-core/ganon/buildcustom/main.nf:44]: Can you add stub? 😆

**Review by nvnieuwk — COMMENTED** [review]
Some small comments, I'll approve when they are addressed :)

File: tests/modules/nf-core/ganon/buildcustom/nextflow.config [hunk 4]
```diff
@@ -0,0 +1,7 @@
+process {
+
+    publishDir = { "${params.outdir}/${task.process.tokenize(':')[-1].tokenize('_')[0].toLowerCase()}" }
+
+    ext.args = '--verbose -x ncbi --write-info-file --ncbi-sequence-info --ncbi-file-info -e fasta'
```
[the remark this answers is not in the document]
Comment (jfy133) [reply tests/modules/nf-core/ganon/buildcustom/nextflow.config:5]: Yes to maximise output files

File: modules/nf-core/ganon/buildcustom/main.nf (hunk 1 above)
Comment (jfy133) [reply modules/nf-core/ganon/buildcustom/main.nf:3]: Fair point!

Comment (jfy133) [inline modules/nf-core/ganon/buildcustom/main.nf:3]:
```suggestion
    label 'process_high'
```

File: modules/nf-core/ganon/buildcustom/main.nf (hunk 3 above)
Comment (jfy133) [reply modules/nf-core/ganon/buildcustom/main.nf:44]: You and your stubs...

Comment (nvnieuwk) [reply modules/nf-core/ganon/buildcustom/main.nf:43]: You'll thank me later hehehehe

File: modules/nf-core/ganon/buildcustom/main.nf [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+        $taxonomy_args \\
+        $genome_size_args \\
+        $args
+
+    cat <<-END_VERSIONS > versions.yml
+    "${task.process}":
+        ganon: \$(echo \$(ganon --version 2>1) | sed 's/.*ganon //g')
+    END_VERSIONS
+    """
+
```
Comment (jfy133) [inline modules/nf-core/ganon/buildcustom/main.nf:43]:
```suggestion

```

File: modules/nf-core/ganon/buildcustom/main.nf (hunk 3 above)
Comment (jfy133) [reply modules/nf-core/ganon/buildcustom/main.nf:58]: Done! does that look OK?

Comment (nvnieuwk) [reply modules/nf-core/ganon/buildcustom/main.nf:58]: Perfect!

**Review by sofstam — APPROVED** [review]

**Pull request merged**